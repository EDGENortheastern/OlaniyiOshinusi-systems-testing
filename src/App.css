* {
  box-sizing: border-box;
  font-family: Arial, Helvetica, sans-serif;
  font-size: large;
}

body {
  background-color: rgb(121, 202, 202);
}

input {
  padding: 0.5rem;
  background-color: beige;
}
