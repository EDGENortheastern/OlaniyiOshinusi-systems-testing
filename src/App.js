import React, {useState} from 'react'
import './App.css';

function App() {

  // state
  const [weight, setWeight] = useState("")
  const [height, setHeight] = useState("")
  const [bmi, setBmi] = useState('')
  const [message, setMessage] = useState('')

  let calcBmi = (event) => {
    //prevent submitting, e=event
    event.preventDefault()

    if (weight === 0 || height === 0) {
      alert('Please enter a valid weight and height')
    } else {
      let bmi = (weight / (height * height))
      setBmi(bmi.toFixed(1)) 

      // Logic for message

      if (bmi < 25) {
        setMessage('You are underweight')
      } else if (bmi >= 25 && bmi < 30) {
        setMessage('You are a healthy weight')
      } else {
        setMessage('You are overweight')
      }
    }
  }
  
  let reload = () => {
    window.location.reload()
  }

  return (
    <div className="app">
      <div className='container'>
        <h1 className='center'>BMI Calculator</h1>
        <form onSubmit={calcBmi}>
          <div>
            <label htmlFor='weight'>Weight (kg)</label>
            <input id="weight" value={weight} onChange={(e) => setWeight(e.target.value)} />
          </div>
          <div>
            <label htmlFor='height'>Height (metres)</label>
            <input id="height" value={height} onChange={(e) => setHeight(e.target.value)} />
          </div>
          <div>
            <button className='btn' type='submit'>Submit</button>
            <button className='btn btn-outline' onClick={reload} type='submit'>Reload</button>
          </div>
        </form>

        <div className='center' role = 'article'>
          <h2>Your BMI is: {bmi}</h2>
          <p data-testid='output'>{message}</p>
        </div>
      </div>        
    </div>
  );
}

export default App;
