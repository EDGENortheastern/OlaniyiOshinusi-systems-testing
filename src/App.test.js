import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

test("the weight input box is in the DOM", () => {
  
  render(<App/>);
  const weightInput = screen.getByLabelText(/weight/i);
  expect(weightInput).toBeInTheDocument();
  
});

test("the height input box is in the DOM", () => {
  
  render(<App/>);
  const heightInput = screen.getByLabelText(/height/i);
  expect(heightInput).toBeInTheDocument();
  
});

test("the weight input box is empty at the start", () => {
  
  render(<App/>);
  const weightInput = screen.getByLabelText(/weight/i);
  expect(weightInput.value).toBe("");
  
});

test("the height input box is empty at the start", () => {
  
  render(<App/>);
  const heightInput = screen.getByLabelText(/height/i);
  expect(heightInput.value).toBe("");
  
});

test("that button submit exists in the DOM", () => {
  
  render(<App/>);
  const btnSubmit = screen.getByText(/submit/i);
  expect(btnSubmit).toBeInTheDocument();
  
});

test("that button reload exists in the DOM", () => {
  
  render(<App/>);
  const btnReload = screen.getByText(/reload/i);
  expect(btnReload).toBeInTheDocument();
  
});

test("that output article exists in the DOM", () => {
  
  render(<App/>);
  const artOutput = screen.getByRole("article");
  expect(artOutput).toBeInTheDocument();
  
});

test("that output paragraph exists in the DOM", () => {
  
  render(<App/>);
  const pOutput = screen.getByTestId("output");
  expect(pOutput).toBeInTheDocument();
  
});
