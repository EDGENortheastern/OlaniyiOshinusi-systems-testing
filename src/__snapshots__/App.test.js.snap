// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks that the App DOM is consistent 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <h1
      className="center"
    >
      BMI Calculator
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="weight"
        >
          Weight (kg)
        </label>
        <input
          id="weight"
          onChange={[Function]}
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="height"
        >
          Height (metres)
        </label>
        <input
          id="height"
          onChange={[Function]}
          value=""
        />
      </div>
      <div>
        <button
          className="btn"
          type="submit"
        >
          Submit
        </button>
        <button
          className="btn btn-outline"
          onClick={[Function]}
          type="submit"
        >
          Reload
        </button>
      </div>
    </form>
    <div
      className="center"
      role="article"
    >
      <h2>
        Your BMI is: 
      </h2>
      <p
        data-testid="output"
      />
    </div>
  </div>
</div>
`;
